apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -74,241 +63,754 @@ static void clean_all_data(int fd, char *buffer)
+{
+	TC_ASSERT_LT("pcm_prepare", pcm_prepare(NULL), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_open_by_name_p
+* @testcase         itc_audio_pcm_open_by_name_p
 * @brief            open with name and initlaize pcm structure
 * @scenario         Open pcm and check its validation
 * @apicovered       pcm_open_by_name
 * @precondition     NA
 * @postcondition    NA
 */
-static void itc_audio_pcm_open_by_name_tc_p(void)
+static void itc_audio_pcm_open_by_name_p(void)
 {
+	g_pcm = NULL;
 	g_pcm = pcm_open_by_name("hw:0,0", PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT_CLEANUP("pcm_open_by_name", pcm_get_file_descriptor(g_pcm), 0, pcm_close(g_pcm))
+
 	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_get_config_p
+* @testcase         itc_audio_pcm_open_by_name_n
+* @brief            open with name and initlaize pcm structure
+* @scenario         Open pcm and with invalid name, check its validation
+* @apicovered       pcm_open_by_name
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_open_by_name_n(void)
+{
+	struct pcm *ptr_pcm = NULL;
+	ptr_pcm = pcm_open_by_name(NULL, PCM_IN, NULL);
+	TC_ASSERT_EQ_CLEANUP("pcm_open", ptr_pcm, 0, pcm_close(g_pcm));
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_get_config_p
 * @brief            get configuration values of pcm
 * @scenario         get configuration values with pcm structure
 * @apicovered       pcm_get_config
 * @precondition     pcm should be opened before.
 * @postcondition    NA
 */
-static void itc_audio_pcm_get_config_tc_p(void)
+static void itc_audio_pcm_get_config_p(void)
 {
 	const struct pcm_config *config = NULL;
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
 	config = pcm_get_config(g_pcm);
 	TC_ASSERT_EQ_CLEANUP("pcm_get_config", config->channels, AUDIO_DEFAULT_CHANNELS, pcm_close(g_pcm));
 	TC_ASSERT_EQ_CLEANUP("pcm_get_config", config->format, AUDIO_DEFAULT_FORMAT, pcm_close(g_pcm));
 	TC_ASSERT_EQ_CLEANUP("pcm_get_config", config->rate, AUDIO_DEFAULT_RATE, pcm_close(g_pcm));
+
 	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_get_channels_p
+* @testcase         itc_audio_pcm_get_config_n_after_close
+* @brief            get configuration values of pcm
+* @scenario         get configuration values with pcm structure after close
+* @apicovered       pcm_get_config
+* @precondition     pcm should be opened and closed before.
+* @postcondition    NA
+*/
+static void itc_audio_pcm_get_config_n_after_close(void)
+{
+	const struct pcm_config *config = NULL;
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
```

Comment:
cleanup needed here.

Reply:
There is no need to close g_pcm, as it is not yet open. This Assert checks whether pcm_open API is actually open or not, if it is not opened, then it cannot be closed